Feat/renewal rating

alex-silverman opened the pull request.

Commit:
rename modules

Commit:
define filtered_data reactive outside edit_coc_version button.

Commit:
add piping_text field to rating_factors. This basically combines the rating text and goal into a comprehensive line about how to evaluate the project against that factor

Commit:
stop converting project_type and target_population to factor in generating test data. By that point, they're already numeric.

Commit:
split rating and rating scores into separate modules

Commit:
adding "rating" tab and renaming TABS_TO_SHOW to TABS_AFTER_LOGIN for clarity

Commit:
adding "selected_criteria" variable and using it among other lgic to dynamically show/hide tabs

Commit:
renaming modules for easier organization

Commit:
adding module_returns. For now, this is how we can talk across modules, since returning values is harder in a loop. This way, we can also set values anywhere and have them accessible everywhere.

also made a number oc hanges to customize rating criteria and rating pages to get things to save and link together.

for now, we're using module_returns to automatically update the factors seen when rating if the user goes back and modifies the customize rating criteria tab. Eventually, we probably want to actually pass this info around, rather than simply triggering a pull from the db. but this was the only way I could figure out to trigger a re-pull

Commit:
edits to rating factcors to capture some of the things in the Excel tool

Commit:
instead of multiplying by -1 to trigger changes to customize_rating_criteria in order to trigger updated factors displayed on the ind rating page, use ! operator, since default is 0 and multiplying by - 1 is still 0.

Commit:
add JS to update group sub totals

also change UI for ind rating page. Instead of nested accordion cards like Customize Rating Criteria, the new UI will look more like the Excel tool with the groups as accordions but then a more tabular like format.

Commit:
add save button to Ind rating page.

Commit:
adding 0 prefix to modules so if we get into double-digits they retain order

Commit:
make clear which tabs are individual rating

Commit:
removing commented out `selected_criteria` which is not necessary in the main rating module

Commit:
another big reorg. this time, we're combining all things rating. this is a more intuitive interface.

Commit:
comments and collapse sidebar when on Customize Rating CRiteria tab

Commit:
convert %>% to |> where possible

Commit:
saving thresholds entries

mlfurman3 reviewed (commented):
File naming is super helpful for keeping modules and sub-modules straight, thanks. 

Clicking around in the Rating tab, I did get it to produce the following bug:

<img width="621" height="337" alt="image" src="https://github.com/user-attachments/assets/7a32f7c3-0203-440d-a37e-a1dc1aa36903" />


<img width="1293" height="318" alt="image" src="https://github.com/user-attachments/assets/b8c5978a-e3bc-44e7-b785-bc3fdef3d20c" />

alex-silverman commented:
> File naming is super helpful for keeping modules and sub-modules straight, thanks.
> 
> Clicking around in the Rating tab, I did get it to produce the following bug:
> 
> <img alt="image" width="621" height="337" src="https://private-user-images.githubusercontent.com/51181259/513369928-7a32f7c3-0203-440d-a37e-a1dc1aa36903.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjI5NzAxNTIsIm5iZiI6MTc2Mjk2OTg1MiwicGF0aCI6Ii81MTE4MTI1OS81MTMzNjk5MjgtN2EzMmY3YzMtMDIwMy00NDBkLWEzN2UtYTFkYzFhYTM2OTAzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTEyVDE3NTA1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThlZTRhMjE5YWM4OGQyZDVjNjU4YjAxMWVkMDk3OTdjOTUwYzQ0M2NlMjdmZWJiZDM2N2E3MDJiNzk1ZGY1ZDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ifr74_jxlMzpg1HuWnrHFhvB7j6LMt3VNwAedbYfxhU"> <img alt="image" width="1293" height="318" src="https://private-user-images.githubusercontent.com/51181259/513369336-b8c5978a-e3bc-44e7-b785-bc3fdef3d20c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjI5NzAxNTIsIm5iZiI6MTc2Mjk2OTg1MiwicGF0aCI6Ii81MTE4MTI1OS81MTMzNjkzMzYtYjhjNTk3OGEtZTNiYy00NGU3LWI3ODUtYmMzZmRlZjNkMjBjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTEyVDE3NTA1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYzMmZiYmM1Yjk5NzAzOGM2N2M5M2ZiZDJkYTgzZmU2MGQxMzI2MDhhNzhmYmQzMzI5NTZhNWEyOTE5YTAzYTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.VCYTaQOMhiSxmoi0qwHrJIIasWFoDKJQEH2JFvimFG0">

do you know how you got to this? Haven't been able to replicate on my end. Do you have no factors selected?

mlfurman3 commented:
Correct, I did not actually enter any Project or Factor data.

1. Edit my instance for AL501
2. Go to Rate -> "Rate Projects" tab and click "In-App" button.
3. Click "Rate/Renewal Expansion Projects" subtab
4. Click "Rating Entry" sub-sub-tab

Commit:
reorganizing rating to customize rating criteria is all together and making sure thresholds are savable to the db

Commit:
changing %>% to |> in a few cases to make sure the code works

Commit:
removing old "is_new" parameter in add_user_stamp function and making the code more readable

Commit:
swapping threshold_id for selected_threshold_id in the threshold_entries table. This way it connected ot both CoC AND HUD thresholds, the latter of which is not selectable.

Also changing table names to be clear that it's CoC Thresholds to be selected.

Commit:
removing generate_sqlite_db. We don't need that anymore. populate_db suffices.

Commit:
code cleanups

Commit:
update module call for rating_scores_entry, the new name

Commit:
open CoC Thresholds block by default on Thresholds entry tab

Commit:
going back to old way of doing add_user_stamp. We only want to replace created_by if it's new.

also, let's use usernname, which will be the email address. therefore, no need for the email field in the user's table

Commit:
anyv doesn't work for some reason when grequesting users.

Commit:
in demo data, 10:13 adds an extra coc_version we don't want

Commit:
better comments about admin user initialization. eventually, when popuklating DB (first and only time) we should dynaically pull from Cognito

Commit:
when creating a new coc, make sure we're checking the value of the button. Otherwise, the Continue? code will run and the modal will immediately disappear

Commit:
make the Request Access to a CoC pop-up a little wider and adjust label text

mlfurman3 committed:
Merge pull request #20 from abtassociates/hotfix/create-new-coc

Hotfix/create new coc

Commit:
add checks for 0-row and NULL data frames in threshold entry and rating scores entry subtabs. this prevents errors when someone has not entered enough project details to populate these objects and clicks around the Rating Scores tab.

mlfurman3 commented:
> Correct, I did not actually enter any Project or Factor data.
> 
> 1. Edit my instance for AL501
> 2. Go to Rate -> "Rate Projects" tab and click "In-App" button.
> 3. Click "Rate/Renewal Expansion Projects" subtab
> 4. Click "Rating Entry" sub-sub-tab

Added a fix for this issue with 0-row and NULL checks. Will merge in to dev.
